src/query/storage/promremote/options.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package promremote
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/uber-go/tally"
+
+	"github.com/m3db/m3/src/cmd/services/m3query/config"
+	xhttp "github.com/m3db/m3/src/x/net/http"
+)
+
+// NewOptions constructs Options based on the given config.
+func NewOptions(cfg *config.PrometheusRemoteBackendConfiguration, scope tally.Scope) (Options, error) {
+	err := validateBackendConfiguration(cfg)
+	if err != nil {
+		return Options{}, err
+	}
+	endpoints := make([]EndpointOptions, len(cfg.Endpoints))
+
+	for i, endpoint := range cfg.Endpoints {
+		endpoints[i] = EndpointOptions{
+			name:    endpoint.Name,
+			address: endpoint.Address,
+		}
+		if endpoint.StoragePolicy != nil {
+			endpoints[i].resolution = endpoint.StoragePolicy.Resolution
```

Comment:
nit: I think it would be more clear if `EndpointOptions` would be constructed and assigned to `endpoints[i]` once.